pkg/utils/grpcutil/grpcutil.go

```diff
@@ -160,3 +171,57 @@ func GetForwardedHost(ctx context.Context) string {
 	}
 	return ""
 }
+
+func establish(ctx context.Context, addr string, tlsConfig *TLSConfig) (*grpc.ClientConn, error) {
+	tlsCfg, err := tlsConfig.ToTLSConfig()
+	if err != nil {
+		return nil, err
+	}
+	cc, err := GetClientConn(
+		ctx,
+		addr,
+		tlsCfg,
+		grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(msgSize)),
+		grpc.WithKeepaliveParams(keepalive.ClientParameters{
+			Time:    keepaliveTime,
+			Timeout: keepaliveTimeout,
+		}),
+		grpc.WithConnectParams(grpc.ConnectParams{
+			Backoff: backoff.Config{
+				BaseDelay:  time.Second,     // Default was 1s.
+				Multiplier: 1.6,             // Default
+				Jitter:     0.2,             // Default
+				MaxDelay:   3 * time.Second, // Default was 120s.
+			},
+			MinConnectTimeout: 5 * time.Second,
```

Comment:
Now that we've extracted conn, how about extracting these parameters as well

Reply:
sounds great